### ISSUE TITLE
FreeBSD, crash and core dump during saving options

### ISSUE THREAD

**Issue by filux** [MegaGlest/megaglest-source#73]
```
megaglest v3.12-dev
Compiled using: Clang: 3.4.1 [64bit] on: Oct  8 2015 12:56:29 platform: FreeBSD-X64 endianness: little
GIT: [Rev: 5170.02fe61e] - using STREFLOP [SSE] - [no-denormals]
```
saving audio options:
```
[New process 100145]
[New process 100256]
[New process 100253]
[New process 100248]
[New process 100247]
[New Thread 81000e800 (LWP 100256)]
[New Thread 80f945c00 (LWP 100253)]
[New Thread 80a408000 (LWP 100248)]
[New Thread 80a407c00 (LWP 100247)]
[New Thread 80a406400 (LWP 100145)]
Core was generated by `megaglest'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00000008019874a0 in _alcDeviceResume () from /usr/local/lib/libopenal.so.0
#0  0x00000008019874a0 in _alcDeviceResume () from /usr/local/lib/libopenal.so.0
#1  0x0000000801985332 in alcMakeContextCurrent () from /usr/local/lib/libopenal.so.0
#2  0x000000000096fd15 in Shared::Sound::OpenAL::SoundPlayerOpenAL::init (this=0x80a7bcf40, params=<optimized out>) at /usr/home/filux/megaglest-git/source/shared_lib/sources/sound/openal/sound_player_openal.cpp:404
#3  0x0000000000783b15 in Glest::Game::SoundRenderer::init (this=0xd96860 <Glest::Game::SoundRenderer::getInstance()::soundRenderer>, window=<optimized out>) at /usr/home/filux/megaglest-git/source/glest_game/sound/sound_renderer.cpp:68
#4  0x00000000006fd666 in Glest::Game::MenuStateOptionsSound::saveConfig (this=0x80a7d4000) at /usr/home/filux/megaglest-git/source/glest_game/menu/menu_state_options_sound.cpp:453
#5  0x00000000006fcd46 in Glest::Game::MenuStateOptionsSound::mouseClick (this=0x80a7d4000, x=<optimized out>, y=<optimized out>, mouseButton=80236394) at /usr/home/filux/megaglest-git/source/glest_game/menu/menu_state_options_sound.cpp:289
#6  0x0000000000585d04 in Glest::Game::MainWindow::eventMouseDown (this=<optimized out>, x=<optimized out>, y=<optimized out>, mouseButton=<optimized out>) at /usr/home/filux/megaglest-git/source/glest_game/main/main.cpp:885
#7  0x000000000092ee62 in Shared::Platform::Window::handleMouseDown (this=0x80a549300, event=...) at /usr/home/filux/megaglest-git/source/shared_lib/sources/platform/sdl/window.cpp:675
#8  0x000000000092d919 in Shared::Platform::Window::handleEvent () at /usr/home/filux/megaglest-git/source/shared_lib/sources/platform/sdl/window.cpp:161
#9  0x00000000005c2b59 in Glest::Game::glestMain (argc=<optimized out>, argv=<optimized out>) at /usr/home/filux/megaglest-git/source/glest_game/main/main.cpp:5454
#10 0x00000000005c3919 in Glest::Game::glestMainSEHWrapper (argc=0, argv=0x20005008) at /usr/home/filux/megaglest-git/source/glest_game/main/main.cpp:5787
#11 0x000000000040e83f in _start ()
```

saving video options:
```
[New process 100157]
[New process 100279]
[New process 100276]
[New process 100271]
[New process 100270]
[New Thread 81000e800 (LWP 100279)]
[New Thread 80f945c00 (LWP 100276)]
[New Thread 80a408000 (LWP 100271)]
[New Thread 80a407c00 (LWP 100270)]
[New Thread 80a406400 (LWP 100157)]
Core was generated by `megaglest'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00000008019874a0 in _alcDeviceResume () from /usr/local/lib/libopenal.so.0
#0  0x00000008019874a0 in _alcDeviceResume () from /usr/local/lib/libopenal.so.0
#1  0x0000000801985332 in alcMakeContextCurrent () from /usr/local/lib/libopenal.so.0
#2  0x000000000096fd15 in Shared::Sound::OpenAL::SoundPlayerOpenAL::init (this=0x80a7bcf40, params=<optimized out>) at /usr/home/filux/megaglest-git/source/shared_lib/sources/sound/openal/sound_player_openal.cpp:404
#3  0x0000000000783b15 in Glest::Game::SoundRenderer::init (this=0xd96860 <Glest::Game::SoundRenderer::getInstance()::soundRenderer>, window=<optimized out>) at /usr/home/filux/megaglest-git/source/glest_game/sound/sound_renderer.cpp:68
#4  0x00000000006efa5d in Glest::Game::MenuStateOptionsGraphics::saveConfig (this=0x80f958000) at /usr/home/filux/megaglest-git/source/glest_game/menu/menu_state_options_graphics.cpp:948
#5  0x00000000006ee318 in Glest::Game::MenuStateOptionsGraphics::mouseClick (this=0x80f958000, x=<optimized out>, y=<optimized out>, mouseButton=80236394) at /usr/home/filux/megaglest-git/source/glest_game/menu/menu_state_options_graphics.cpp:650
#6  0x0000000000585d04 in Glest::Game::MainWindow::eventMouseDown (this=<optimized out>, x=<optimized out>, y=<optimized out>, mouseButton=<optimized out>) at /usr/home/filux/megaglest-git/source/glest_game/main/main.cpp:885
#7  0x000000000092ee62 in Shared::Platform::Window::handleMouseDown (this=0x80a549300, event=...) at /usr/home/filux/megaglest-git/source/shared_lib/sources/platform/sdl/window.cpp:675
#8  0x000000000092d919 in Shared::Platform::Window::handleEvent () at /usr/home/filux/megaglest-git/source/shared_lib/sources/platform/sdl/window.cpp:161
#9  0x00000000005c2b59 in Glest::Game::glestMain (argc=<optimized out>, argv=<optimized out>) at /usr/home/filux/megaglest-git/source/glest_game/main/main.cpp:5454
#10 0x00000000005c3919 in Glest::Game::glestMainSEHWrapper (argc=0, argv=0x20005008) at /usr/home/filux/megaglest-git/source/glest_game/main/main.cpp:5787
#11 0x000000000040e83f in _start ()
```

saving network options:
```
[New process 100145]
[New process 100268]
[New process 100259]
[New process 100258]
[New Thread 81000e800 (LWP 100268)]
[New Thread 80a408000 (LWP 100259)]
[New Thread 80a407c00 (LWP 100258)]
[New Thread 80a406400 (LWP 100145)]
Core was generated by `megaglest'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00000008019874a0 in _alcDeviceResume () from /usr/local/lib/libopenal.so.0
#0  0x00000008019874a0 in _alcDeviceResume () from /usr/local/lib/libopenal.so.0
#1  0x0000000801985332 in alcMakeContextCurrent () from /usr/local/lib/libopenal.so.0
#2  0x000000000096fd15 in Shared::Sound::OpenAL::SoundPlayerOpenAL::init (this=0x80a7bcf40, params=<optimized out>) at /usr/home/filux/megaglest-git/source/shared_lib/sources/sound/openal/sound_player_openal.cpp:404
#3  0x0000000000783b15 in Glest::Game::SoundRenderer::init (this=0xd96860 <Glest::Game::SoundRenderer::getInstance()::soundRenderer>, window=<optimized out>) at /usr/home/filux/megaglest-git/source/glest_game/sound/sound_renderer.cpp:68
#4  0x00000000006f7ae1 in Glest::Game::MenuStateOptionsNetwork::saveConfig (this=0x80a5e7000) at /usr/home/filux/megaglest-git/source/glest_game/menu/menu_state_options_network.cpp:529
#5  0x00000000006f7201 in Glest::Game::MenuStateOptionsNetwork::mouseClick (this=0x80a5e7000, x=<optimized out>, y=<optimized out>, mouseButton=80236394) at /usr/home/filux/megaglest-git/source/glest_game/menu/menu_state_options_network.cpp:347
#6  0x0000000000585d04 in Glest::Game::MainWindow::eventMouseDown (this=<optimized out>, x=<optimized out>, y=<optimized out>, mouseButton=<optimized out>) at /usr/home/filux/megaglest-git/source/glest_game/main/main.cpp:885
#7  0x000000000092ee62 in Shared::Platform::Window::handleMouseDown (this=0x80a549300, event=...) at /usr/home/filux/megaglest-git/source/shared_lib/sources/platform/sdl/window.cpp:675
#8  0x000000000092d919 in Shared::Platform::Window::handleEvent () at /usr/home/filux/megaglest-git/source/shared_lib/sources/platform/sdl/window.cpp:161
#9  0x00000000005c2b59 in Glest::Game::glestMain (argc=<optimized out>, argv=<optimized out>) at /usr/home/filux/megaglest-git/source/glest_game/main/main.cpp:5454
#10 0x00000000005c3919 in Glest::Game::glestMainSEHWrapper (argc=0, argv=0x20005008) at /usr/home/filux/megaglest-git/source/glest_game/main/main.cpp:5787
#11 0x000000000040e83f in _start ()
```

**Comment by softcoder** [comment]
please try this with the ini setting:

DebugSound=true

and paste output (use a new build)

**Comment by filux** [comment]
closed in waiting for changes related with #105

**Comment by filux** [comment]
Based on the https://github.com/MegaGlest/megaglest-source/issues/105#issuecomment-257027347, this issue is not reproducible on v3.12.0 (based on later commit than my tests were done), so issue will stay closed.
